src/dash-table/derived/cell/dropdowns.ts

```diff
@@ -1,94 +1,128 @@
+                        dropdown_properties[column.id][realIndex] :
+                        null
+                )
+            ) || column
+        ).options;
 
-    const conditional = columnConditionalDropdown.find((cs: any) => cs.id === column.id);
-    const base = columnStaticDropdown.find((ss: any) => ss.id === column.id);
+        const conditional = columnConditionalDropdown.find((cs: any) => cs.id === column.id);
+        const base = columnStaticDropdown.find((ss: any) => ss.id === column.id);
 
-    const conditionalDropdowns = (conditional && conditional.dropdowns) || [];
-    const staticDropdown = legacyDropdown || (base && base.dropdown);
+        return [
+            legacyDropdown || (base && base.dropdown),
+            (conditional && conditional.dropdowns) || []
+        ];
+    });
 
-    return getDropdown(astCache, conditionalDropdowns, datum, column.id, staticDropdown);
-}, columns), data);
+    /**
+     * Returns the highest priority dropdown from the
+     * applicable dropdowns.
+     */
+    private readonly dropdown = memoizerCache<[ColumnId, number]>()((
+        applicableDropdowns: [any, any],
+        column: IVisibleColumn,
+        datum: Datum
+    ) => {
+        const [staticDropdown, conditionalDropdowns] = applicableDropdowns;
 
-const getterFactory = memoizeOneFactory(getter);
+        const matches = [
+            ...(staticDropdown ? [staticDropdown] : []),
+            ...R.map(
+                ([cd]) => cd.dropdown,
+                R.filter(
+                    ([cd, i]) => this.evaluation.get(column.id, i)(
+                        this.ast.get(column.id, i)(cd.condition),
+                        datum
+                    ),
+                    R.addIndex<IConditionalDropdown, [IConditionalDropdown, number]>(R.map)(
+                        (cd, i) => [cd, i],
+                        conditionalDropdowns
+                    ))
+            )
+        ];
 
-const decoratedGetter = (_id: string): ((
-    columns: VisibleColumns,
-    data: Data,
-    indices: Indices,
-    columnConditionalDropdown: any,
-    columnStaticDropdown: any,
-    dropdown_properties: any
-) => (DropdownValues | undefined)[][]) => {
-    const astCache = memoizerCache<[ColumnId, number], [string], SyntaxTree>(
-        (query: string) => new SyntaxTree(query)
-    );
+        return matches.length ? matches.slice(-1)[0] : undefined;
+    });
 
-    return getterFactory().bind(undefined, astCache);
-};
+    /**
+     * Get the query's AST.
+     */
+    private readonly ast = memoizerCache<[ColumnId, number]>()((
+        query: string
+    ) => new SyntaxTree(query));
 
-export default memoizeOneFactory(decoratedGetter);
+    /**
+     * Evaluate if the query matches the cell's data.
+     */
+    private readonly evaluation = memoizerCache<[ColumnId, number]>()((
+        ast: SyntaxTree,
+        datum: Datum
+    ) => ast.evaluate(datum));
```

Comment:
Caching both the SyntaxTree generated from the filter condition and the result of a given tree evaluation. As per above, this is for each cell.